fish-rust/src/wgetopt.rs

```diff
@@ -21,41 +23,39 @@ License along with the GNU C Library; see the file COPYING.LIB.  If
 not, write to the Free Software Foundation, Inc., 675 Mass Ave,
 Cambridge, MA 02139, USA.  */
 
-/// Note wgetopter expects an mutable array of const strings. It modifies the order of the
-/// strings, but not their contents.
 use crate::wchar::{utf32str, wstr, WExt};
 
-// Describe how to deal with options that follow non-option ARGV-elements.
-//
-// If the caller did not specify anything, the default is PERMUTE.
-//
-// REQUIRE_ORDER means don't recognize them as options; stop option processing when the first
-// non-option is seen. This is what Unix does. This mode of operation is selected by using `+'
-// as the first character of the list of option characters.
-//
-// PERMUTE is the default.  We permute the contents of ARGV as we scan, so that eventually all
-// the non-options are at the end.  This allows options to be given in any order, even with
-// programs that were not written to expect this.
-//
-// RETURN_IN_ORDER is an option available to programs that were written to expect options and
-// other ARGV-elements in any order and that care about the ordering of the two.  We describe
-// each non-option ARGV-element as if it were the argument of an option with character code 1.
-// Using `-' as the first character of the list of option characters selects this mode of
-// operation.
-//
-// The special argument `--' forces an end of option-scanning regardless of the value of
-// `ordering'.  In the case of RETURN_IN_ORDER, only `--' can cause `getopt' to return EOF with
-// `woptind' != ARGC.
+/// Describe how to deal with options that follow non-option ARGV-elements.
+///
+/// If the caller did not specify anything, the default is PERMUTE.
+///
+/// REQUIRE_ORDER means don't recognize them as options; stop option processing when the first
+/// non-option is seen. This is what Unix does. This mode of operation is selected by using `+'
+/// as the first character of the list of option characters.
+///
+/// PERMUTE is the default.  We permute the contents of ARGV as we scan, so that eventually all
+/// the non-options are at the end.  This allows options to be given in any order, even with
+/// programs that were not written to expect this.
+///
+/// RETURN_IN_ORDER is an option available to programs that were written to expect options and
+/// other ARGV-elements in any order and that care about the ordering of the two.  We describe
+/// each non-option ARGV-element as if it were the argument of an option with character code 1.
+/// Using `-` as the first character of the list of option characters selects this mode of
+/// operation.
+///
+/// The special argument `--` forces an end of option-scanning regardless of the value of
+/// `ordering`.  In the case of RETURN_IN_ORDER, only `--` can cause `getopt` to return EOF with
+/// `woptind` != ARGC.
 #[derive(Debug, Clone, Copy, PartialEq, Eq)]
 enum Ordering {
-    REQUIRE_ORDER,
-    PERMUTE,
-    RETURN_IN_ORDER,
+    RequireOrder,
```

Comment:
The docstring and the C++ dual still use screaming snake case.

Keeping the existing name is the easiest way to port code but I think we want to adopt Rust conventions.

I have no strong opinion yet on whether to do this now or later (when the port is done).

I'd like to experiment a bit more before deciding this, so maybe instruct clippy to silence this warning?

Reply:
Silenced.